### PR TITLE
add object and object storage testcase

### PR DESCRIPTION
**Pull request by baichuan3**
part of https://github.com/rooch-network/rooch/issues/193

### CODE REVIEW

File: moveos/moveos-stdlib/moveos-stdlib/sources/object_storage.move [hunk 1]
```diff
@@ -108,4 +108,50 @@ module moveos_std::object_storage {
...
+    #[expected_failure]
+    fun test_borrow_not_exist_failure(sender: signer) {
+        let sender_addr = std::signer::address_of(&sender);
+        let ctx = moveos_std::tx_context::new_test_context(sender_addr);
+        let os = new(&mut ctx);
+        let object = object::new(&mut ctx, sender_addr, TestObject{f: 1});
+        let object_id = object::id(&object);
+
+        let _obj_ref = borrow<TestObject>(&os, object_id);
+        abort 21
```
Comment (jolestar) [inline moveos/moveos-stdlib/moveos-stdlib/sources/object_storage.move:122]: Why do we need to abort here? I think the `borrow` function should abort.

File: crates/rooch-framework-tests/tests/cases/object/object_cap.move [hunk 2]
```diff
@@ -27,9 +31,10 @@ script {
         let object = m::new_test_object(12);
         let obj = object::new<TestObject>(storage_context::tx_context_mut(ctx), sender_addr, object);
 
-        let _borrow_object = object::borrow_mut(&mut obj);
-        abort 21
-
+        let _borrow_object = object::borrow(&obj);
```
Comment (jolestar) [inline crates/rooch-framework-tests/tests/cases/object/object_cap.move:34]: if the #327 merged, this test case should fail with compile error status

**Review by jolestar — APPROVED** [review]
@steelgeek091 if this PR merge first, you need to rebase and update the test case.

**Comment by steelgeek091** [comment]
> 

Okay，Can my PR be merged first?

**Comment by baichuan3** [comment]
After merge, the private_generics verify work well, @steelgeek091 @jolestar 

```
task 2 'run'. lines 22-39:
error: resource type "TestObject" in function "0x2::object::new" not defined in current module or not allowed
   ┌─ /var/folders/b7/gt6_vlj14mn6h8xk1x9ly6m40000gn/T/.tmpUaDNWj:31:19
   │
31 │         let obj = object::new<TestObject>(storage_context::tx_context_mut(ctx), sender_addr, object);
   │                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: resource type "TestObject" in function "0x2::object::borrow" not defined in current module or not allowed
   ┌─ /var/folders/b7/gt6_vlj14mn6h8xk1x9ly6m40000gn/T/.tmpUaDNWj:32:30
   │
32 │         let _borrow_object = object::borrow(&obj);
   │                              ^^^^^^^^^^^^^^^^^^^^

error: resource type "TestObject" in function "0x2::object::unpack" not defined in current module or not allowed
   ┌─ /var/folders/b7/gt6_vlj14mn6h8xk1x9ly6m40000gn/T/.tmpUaDNWj:33:42
   │
33 │         let (_id, _owner, test_object) = object::unpack(obj);
   │                                          ^^^^^^^^^^^^^^^^^^^

status EXECUTED
```

**Comment by baichuan3** [comment]
The next question is, does the integration test now support '// check: ABORTED' like Starcoin style, or 'expected_failure' annotation syntax? @jolestar

**Comment by jolestar** [comment]
The integration test only compare the output with exp file, The starcoin annotation is just for human readings.

**Pull request merged**